avm/res/data-factory/factory/main.bicep

```
@@ -586,3 +587,54 @@ type customerManagedKeyType = {
   @description('Optional. User assigned identity to use when fetching the customer managed key. Required if no system assigned identity is available for use.')
   userAssignedIdentityResourceId: string?
 }?
+
+type managedPrivateEndpointsType = {
```

Sorry @clintgrove if I chime in on one point, I'll leave the rest of the PR review to you 🙂 

I think we are unofficially following a naming convention in singular form for types. If you agree I'd change this and others for consistency
```suggestion
type managedPrivateEndpointType = {
```

---

Hey @eriqua, 
this would be new to be, but can be done. 
Personally, I handled it so far usually as per the following 'rules':
- If the type is singular, the name should be singular (e.g., if the PE type only discribes a single PE)
- If the type is plural (e.g., an array), the name should be plural (e.g., if the PE type describes an array of PEs).
Now in this case, the managed PE type is a an type that's automatically an array.

Maybe we should define this, what do you say @eriqua? For example, our spec-types are in fact all singular, even through they describe a plural instance.

As I'd anyways like to discuss this with you and the team (having types always be singular not only in name but also implementation), I'd suggest to have a chat some time and put if on the agenda if you'd agree? The good news is that the entire point is neither a minor nor major change and completely invisible to the consumer. So the stakes are low 😄

---

Yes, the specs-types one is the reason why I'm so far unofficially following that convention.
I'd suggest going always with the singular one as easier to follow and comment on.

---

sorry guys. I am here now. I shall take a look today

---

I would agree to make is singular semantics

---

done